src/components/banner/Banner.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+import Box from '../box';
+import Island, { IslandProps } from '../island';
+import Section, { SectionProps } from '../section';
+import IconButton from '../iconButton';
+import theme from './theme.css';
+import { IconCloseMediumOutline } from '@teamleader/ui-icons';
+
+export interface BannerProps extends SectionProps, IslandProps {
+  /** The content to display inside the banner. */
+  children?: React.ReactNode;
+  /** A class name for the wrapper to give custom styles. */
+  className?: string;
+  /** The icon displayed on the left inside the banner. */
+  icon?: React.ReactElement;
+  /** Callback function that is fired when the close button of the banner is clicked. */
+  onClose?(...args: unknown[]): unknown;
```

Comment:
Q: why is this unknown arguments and unknown return type?

Reply:
> Q: why is this unknown arguments and unknown return type?

Fixed in follow-up: https://github.com/teamleadercrm/ui/pull/2399